syncd/SwitchNotifications.cpp

```diff
@@ -1,6 +1,7 @@
 #include "SwitchNotifications.h"
 
 #include "swss/logger.h"
+#include <array>
```

Comment:
Why this isnadded?

Reply:
GCC 12 complained that `std::array` was not completely defined, but is instantiated on line 144.

Snippet of error log:

```
SwitchNotifications.cpp: In instantiation of 'constexpr auto declare_static(std::index_sequence<i ...>) [with B = syncd::SwitchNotifications::SlotBase; D = syncd::SwitchNotifications::Slot; long unsigned int ...i = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15}; std::index_sequence<i ...> = std::integer_sequence<long unsigned int, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15>]':
SwitchNotifications.cpp:151:35:   required from 'constexpr auto declare_static() [with B = syncd::SwitchNotifications::SlotBase; D = syncd::SwitchNotifications::Slot; long unsigned int size = 16]'
SwitchNotifications.cpp:156:83:   required from here
SwitchNotifications.cpp:144:56: error: invalid use of incomplete type 'struct std::array<syncd::SwitchNotifications::SlotBase*, 16>'
  144 |     return std::array<B*, sizeof...(i)>{{new D<i>()...}};
      |                                                        ^
In file included from /usr/include/c++/12/bits/stl_map.h:63,
                 from /usr/include/c++/12/map:61,
                 from /usr/include/swss/logger.h:6,
                 from SwitchNotifications.h:7,
                 from SwitchNotifications.cpp:1:
/usr/include/c++/12/tuple:1595:45: note: declaration of 'struct std::array<syncd::SwitchNotifications::SlotBase*, 16>'
...
```